src/acl.c

```diff
@@ -1609,6 +1610,7 @@ sds ACLLoadFromFile(const char *filename) {
             u = ACLGetUserByName(argv[1],sdslen(argv[1]));
             serverAssert(u != NULL);
             ACLSetUser(u,"reset",-1);
+            u->flags = u->flags | server.acl_pubsub_default;
```

Comment:
@huangzhw do you think maybe it would be better to move this into `ACLSetUser("reset")`?

the docs say this about "reset":
>  The user returns to the same state it has immediately after its creation.

Reply:
@oranagra Thie will be a bigger step. If we add this in `reset`, the behavior will not be compatible with 6.2. Although it will conform to the docs.

---------------------------------------------------
Another way this bug affect not `default` user:
```
127.0.0.1:6380> auth hello hello
OK
127.0.0.1:6380> publish hello world
(integer) 0
127.0.0.1:6380> acl load
OK
127.0.0.1:6380> publish hello world
(error) NOPERM this user has no permissions to access one of the channels used as arguments
```

Reply:
I think this is a bug we must fix.
it'll change the (buggy) behavior, but i don't see any other way.
i **don't** think we can document that buggy behavior and keep it (your example above is a clear indication of that).

if we did it'll look something like
> when creating a new user that didn't exist before, `acl-pubsub-default` is respected, but when overriding an existing user, or the default user, all channels are denied unless explicitly allowed.

same goes for `reset`:
>   reset        Performs the following actions: resetpass, resetkeys, off, **resetchannels**, 
               -@all. The user returns to the same state it has immediately
               after its creation (**except for publish channels that would be denied even when acl-pubsub-default allows them**) 


this makes no sense.

Reply:
@huangzhw , while we're waiting for the core team's approval, if you agree that `reset` should do that, let's make that change in the PR.
i think it'll conform with the docs, and also be more compatible with redis 6.0.

I suppose something like this should do the trick:
```diff
         serverAssert(ACLSetUser(u,"resetkeys",-1) == C_OK);
         serverAssert(ACLSetUser(u,"resetchannels",-1) == C_OK);
+        if (server.acl_pubusub_default & USER_FLAG_ALLCHANNELS)
+            serverAssert(ACLSetUser(u,"allchannels",-1) == C_OK);
```

Reply:
@oranagra Agree. Change made. And I add more tests.